source/security/authentication.txt

```diff
@@ -0,0 +1,545 @@
+         client = pymongo.MongoClient(uri)
+
+.. tip:: AWS Lambda
+   
+   AWS Lambda runtimes can automatically set these environment variables during
+   initialization. For more information about using environment variables in an AWS Lambda
+   environment, see
+   `Using Lambda environment variables <https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime>`__
+   in the AWS documentation.
+
+Shared Credentials File
+~~~~~~~~~~~~~~~~~~~~~~~
+
+If {+driver-short+} doesn't find AWS credentials in the preceding environment variables,
+it tries to read them from a shared credentials file.
+
+To use a shared credentials file to authenticate your application, ensure that the
+file exists in your environment and is configured correctly. To learn how to create
+a shared credentials file, see
+`Credentials <https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#shared-credentials-file>`__
+in the Boto3 documentation and `Configuration <https://docs.aws.amazon.com/sdkref/latest/guide/creds-config-files.html>`__
+in the AWS documentation.
+
+After you create the shared credentials file, set the ``authMechanism``
+connection option to ``"MONGODB-AWS"``.
+You can set this option in two ways: by passing an argument to the
+``MongoClient`` constructor or through a parameter in your connection string.
+
+.. tabs::
+
+   .. tab:: MongoClient
+      :tabid: mongoclient
+
+      .. code-block:: python
+
+         client = pymongo.MongoClient("mongodb://<hostname>:<port>",
+                                      authMechanism="MONGODB-AWS")
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      .. code-block:: python
+
+         uri = "mongodb://<hostname>:<port>/?&authMechanism=MONGODB-AWS"
+         client = pymongo.MongoClient(uri)
+
+.. tip::
+
+   To prevent {+driver-short+} from using a shared credentials file for authentication,
+   perform one of the following actions:
+   
+   - Set the ``AWS_SHARED_CREDENTIALS_FILE`` environment variable to ``""`` in your terminal.
+   - Add ``os.environ["AWS_SHARED_CREDENTIALS_FILE"] = ""`` to your script or
+     application.
+   - Create an AWS profile specifically for your MongoDB credentials. Set the ``AWS_PROFILE``
+     environment variable to the name of the profile you created.
+
+AWS Config File
+~~~~~~~~~~~~~~~
+
+If {+driver-short+} doesn't find credentials in the shared credentials file, it tries to read
+them from an AWS config file.
+
+To use a shared credentials file to authenticate your application, ensure that the
+file exists in your environment and is configured correctly. To learn how to create
+a shared credentials file, see
+`Credentials <https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#aws-config-file>`__
+in the Boto3 documentation and `Configuration <https://docs.aws.amazon.com/sdkref/latest/guide/creds-config-files.html>`__
+in the AWS documentation.
```

Comment:
Was this paragraph purposefully duplicated here?

Reply:
Whoops. It's actually identical except for the link (which I did update) and the first clause (which I did not)